---
title: "wiki"
author: "Thanh Le"
date: "2024-05-30"
output: html_document
---

# Data Processing

For some summary statistics, only the position of the SNPs is provided. Retrieving rsID by aligning the position against the reference SNP database turns out to be a resource consuming and low output task, due to the large size of the reference SNP dataset, for example 1000GENOMES-phase_3.vcf of nearly 10Gb retrieved from Ensemble which has been used for imputation in GWASs.

A more efficient way is to align the positions of the SNPs in sumstat files against the position in LD score files, retrieve rsID from that, and compare the rsID from summary statistics against the reference SNP panel. The bash script `maptoLDhm3.sh` was used for this.
The summary statistics are reformulated to be appropriate for this script.

```{bash eval=FALSE}
cd data/inuse/
awk 'BEGIN {FS=OFS="\t"} {print $10,$11,$7,$5,$6,$2,$1":"$2}' ../sumstats_raw/stroke.tsv > stroke_streamed.txt
sed -i '1s/.*/A1\tA2\tOR\tStdErr\tP-value\tFreqA1\tCHR:BP/' stroke_streamed.txt
../../src/maptoLDhm3.sh ../reference/eur_w_ld_chr/LDscores_concatenated_CHR:BP.txt stroke_streamed.txt ../reference/eur_w_ld_chr/w_hm3_copy.snplist
mv maptold_hm3.txt stroke_maptold_hm3.txt

awk 'BEGIN {FS=OFS="\t"} {print $3,$4,$5,$6,$8,$7,$1":"$2}' ../sumstats_raw/MAGIC_postchallengeIR_IFC_adjBMI_EUR.tsv > IFC_streamed.txt
sed -i '1s/.*/A1\tA2\tBeta\tStdErr\tP-value\tNeff\tCHR:BP/' IFC_streamed.txt
../../src/maptoLDhm3.sh ../reference/eur_w_ld_chr/LDscores_concatenated_CHR:BP.txt IFC_streamed.txt ../reference/eur_w_ld_chr/w_hm3_copy.snplist
mv maptold_hm3.txt IFC_maptold_hm3.txt

awk 'BEGIN {FS=OFS="\t"} {print toupper($4),toupper($5),$10,$11,$12,$19,$22}' ../sumstats_raw/cad.tsv > cad_streamed.txt
sed -i '1s/.*/A1\tA2\tBeta\tStdErr\tP-value\tNeff\tCHR:BP/' cad_streamed.txt
../../src/maptoLDhm3.sh LDscores_concatenated_CHR:BP.txt cad_streamed.txt w_hm3_copy.snplist
mv maptold_hm3.txt cad_maptold_hm3.txt
```


